src/celeritas/field/FieldPropagator.hh

```diff
@@ -180,6 +180,7 @@ CELER_FUNCTION auto FieldPropagator<DriverT, GTV>::operator()(real_type step)
         // Advance up to (but probably less than) the remaining step length
         DriverResult substep = driver_.advance(remaining, state_);
         CELER_ASSERT(substep.step > 0 && substep.step <= remaining);
+        ++result.num_substeps;
```

Comment:
One other useful diagnostic would be to use the "diagnostic substepper" (or equivalent) to count the number of *integrations* as a function of energy.